app/services/recurring_contribution/subscriptions/cancel_expired.rb

```diff
@@ -0,0 +1,38 @@
+class RecurringContribution::Subscriptions::CancelExpired
+  attr_reader :subscriptions
+
+  def initialize(subscriptions)
+    @subscriptions = subscriptions
+  end
+
+  def self.process(subscriptions)
+    new(subscriptions).process
+  end
+
+  def process
+    pagarme_subscriptions = Pagarme::API.retrieve_pagarme_subscriptions(subscriptions.keys)
+    canceled_subscriptions = cancelable_subscriptions(pagarme_subscriptions, subscriptions)
+
+    return if canceled_subscriptions.empty?
+    Pagarme::API.cancel_subscriptions(canceled_subscriptions)
+  end
+
+  private
+
+  def cancelable_subscriptions(pagarme_subscriptions, subscriptions)
+    [].tap do |canceled_subscriptions|
+      pagarme_subscriptions.each do |pagarme_subscription|
+        next if pagarme_subscription[:errors].present?
+
+        subscription = subscriptions[pagarme_subscription[:id]]
+
+        canceled_subscriptions << subscription if cancelable?(pagarme_subscription[:charges], subscription)
+      end
+    end
+  end
+
+  def cancelable?(pagarme_charges, subscription)
+    (subscription.bank_billet? && pagarme_charges == subscription.charges) ||
```

Comment:
Could you break these conditionals into methods to make easier to read this part?